onnxscript/ir/_external_data.py

```diff
@@ -51,3 +70,245 @@ def set_base_dir(graph: _core.Graph | _core.GraphView, base_dir: str | os.PathLi
+
+    Args:
+        external_data_info: A collection of external data information stored for each tensor to be written as external data.
+        file_path: Location to which external data is to be stored.
+    """
+    with open(file_path, "w+b") as data_file:
+        for value, tensor_info in external_data_info:
+            current_offset = tensor_info.offset
+            assert value.const_value is not None
+            # If external tensor, retrieve raw_data from source_file
+            if isinstance(value.const_value, _core.ExternalTensor):
+                external_tensor = value.const_value
+                external_tensor_file_path = os.path.join(
+                    external_tensor.base_dir, external_tensor.path
+                )
+                # If source file for external tensor is the same as the destination path
+                # use the previously renamed file (base_dir/temp/path.bin)
+                if external_tensor_file_path == file_path:
+                    external_tensor.base_dir = external_tensor.base_dir + "/temp/"  # type: ignore[operator]
+                raw_data = external_tensor.tobytes()
+            else:
+                raw_data = value.const_value.tobytes()
+            if current_offset is not None:
+                # Pad file to required offset if needed
+                file_size = data_file.tell()
+                if current_offset > file_size:
+                    data_file.write(b"\0" * (current_offset - file_size))
+            data_file.write(raw_data)
+
+
+def _store_as_external_tensors(
+    model: _core.Model,
+    external_data_info: list[tuple[_core.Value, ExternalDataInfo]],
+    file_path: str | os.PathLike,
+) -> _core.Model:
+    """
+    Convert the tensors (stored within the values) written as external data to _core.ExternalTensor types.
+
+    Args:
+        model: Model to be converted.
+        external_data_info: A collection of external data information stored for each tensor to be written as external data.
+        file_path: Location to which external data is to be stored.
+    """
+    for value, tensor_info in external_data_info:
+        tensor = value.const_value
+        assert tensor is not None
+        external_tensor = _core.ExternalTensor(
+            file_path,
+            tensor_info.offset,
+            tensor_info.length,
+            tensor.dtype,  # type: ignore[arg-type]
+            shape=tensor.shape,  # type: ignore[arg-type]
+            name=tensor.name,  # type: ignore[arg-type]
+        )
+        value.const_value = external_tensor
+        assert value.name in model.graph.initializers
+        model.graph.initializers[value.name] = value
+    return model
+
+
+def to_external_data(
+    model: _core.Model,
+    base_path: str | os.PathLike,
+    file_path: str = "",
+    align_offset: bool = True,
+    align_threshold: int = 1048576,  # 1MB,
+    allocation_granularity: int = 65536,  # 64KB
```

Comment:
```suggestion
```

I think for this function we can just hard code these values (I would move as internal global constants, to keep the interface simple We can add them as options when there is a need.